### PR TITLE
(#98) Fix running bolt tasks on Windows

### LINKED ISSUE #98: Windows: Cannot execute Bolt tasks as the node does not meet the minimum compatability requirements
I have 4 hosts, 3 linux and 1 windows. 

All puppet agents are 5.5.4 (running mcollectived (2.12.2) server, not choria server if that matters)

Puppet Server is 5.3.0

Going through documentation trying to get tasks working and I'm running the command: 

 mco tasks run mcollective_agent_bolt_tasks::ping --message "Hello"

The 3 linux hosts work fine, but the windows hosts returns an error:

```
Retrieving task metadata for task mcollective_agent_bolt_tasks::ping from the Puppet Server
Attempting to download and run task mcollective_agent_bolt_tasks::ping on 4 nodes

Downloading and verifying 1 file(s) from the Puppet Server to all nodes: ✓  4 / 4

Running task mcollective_agent_bolt_tasks::ping and waiting up to 60 seconds for it to complete

<winhost>       Cannot execute Bolt tasks as the node does not meet the minimum compatability requirements



Summary for task e540262f1adb5657b1f660c9ac990aa4

                       Task Name: mcollective_agent_bolt_tasks::ping
                          Caller: choria=mco.mcollective
                       Completed: 3
                         Running: 0
                 Wrapper Failure: 1

                      Successful: 3
                          Failed: 1

                Average Run Time: 0.14s
```

I am wondering if Windows is not supported for tasks or if this agent only works with Choria Server running on the node.

Error logs in mcollective daemon say the following:

```
I, [2018-08-20T14:27:56.145315-04:00 #1032]  INFO -- : runner.rb:216:in `agentmsg' Handling message e540262f1adb5657b1f660c9ac990aa4 for agent 'bolt_tasks' in collective 'mcollective' from choria=mco.mcollective@<client>: {"agent"=>"bolt_tasks", "action"=>"run_and_wait", "caller"=>"choria=mco.mcollective", "data"=>{"task"=>"mcollective_agent_bolt_tasks::ping", "files"=>"[{\"filename\":\"ping.rb\",\"sha256\":\"328716f02f7e45faf53ec842af2e6893af7cd29f6b7de9f69f0a6aeeaa417c9a\",\"size_bytes\":123,\"uri\":{\"path\":\"/puppet/v3/file_content/tasks/mcollective_agent_bolt_tasks/ping.rb\",\"params\":{\"environment\":\"production\"}}}]", "input"=>"{\"message\":\"Hello\"}"}}
E, [2018-08-20T14:27:56.160914-04:00 #1032] ERROR -- : agent.rb:103:in `rescue in handlemsg' bolt_tasks#run_and_wait failed: MCollective::UnknownRPCError: Cannot execute Bolt tasks as the node does not meet the minimum compatability requirements
E, [2018-08-20T14:27:56.160914-04:00 #1032] ERROR -- : agent.rb:104:in `rescue in handlemsg' C:/Program Files/Puppet Labs/Puppet/mcollective/lib/mcollective/rpc/reply.rb:60:in `fail!'
	C:/ProgramData/PuppetLabs/mcollective/plugins/mcollective/agent/bolt_tasks.rb:31:in `block in <class:Bolt_tasks>'
	C:/Program Files/Puppet Labs/Puppet/mcollective/lib/mcollective/rpc/agent.rb:86:in `handlemsg'
	C:/Program Files/Puppet Labs/Puppet/mcollective/lib/mcollective/agents.rb:126:in `block (2 levels) in dispatch'
	C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/2.4.0/timeout.rb:93:in `block in timeout'
	C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/2.4.0/timeout.rb:33:in `block in catch'
	C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/2.4.0/timeout.rb:33:in `catch'
	C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/2.4.0/timeout.rb:33:in `catch'
	C:/Program Files/Puppet Labs/Puppet/sys/ruby/lib/ruby/2.4.0/timeout.rb:108:in `timeout'
	C:/Program Files/Puppet Labs/Puppet/mcollective/lib/mcollective/agents.rb:125:in `block in dispatch'
```

### PR DESCRIPTION
**Pull request by smortex**
I took some time to continue digging in running tasks on windows (again).

Running a Ruby task:

```sh-session
romain@optiplex-3050 ~ % mco tasks run choria::ping --message=w00t -I /desktop-bpi/ -v  
Retrieving task metadata for task choria::ping from the Puppet Server
Discovering hosts using the mc method for 2 second(s) .... 1
Attempting to download and run task choria::ping on 1 nodes

Downloading and verifying 1 file(s) from the Puppet Server to all nodes: ✓  1 / 1

Running task choria::ping and waiting up to 60 seconds for it to complete

desktop-bpi073d.lan
   {"message":"w00t","timestamp":"2021-06-22 13:12:17 -1000"}



Summary for task 5a5e46d98cb05f18ae0853f097a9b346

                       Task Name: choria::ping
                          Caller: choria=romain.mcollective
                       Completed: 1
                         Running: 0

                      Successful: 1
                          Failed: 0

                Average Run Time: 0.14s


romain@optiplex-3050 ~ %
```

Running a PowerShell task (from the `puppetlas-exec` module):

```sh-session
romain@optiplex-3050 ~ % mco tasks run exec --command='echo w00t' -I /desktop-bpi/ -v                      
Retrieving task metadata for task exec from the Puppet Server
Discovering hosts using the mc method for 2 second(s) .... 1
Attempting to download and run task exec on 1 nodes

Downloading and verifying 3 file(s) from the Puppet Server to all nodes: ✓  1 / 1

Running task exec and waiting up to 60 seconds for it to complete

desktop-bpi073d.lan
   {"_output":"w00t\n","exit_code":0}



Summary for task 54690ef6afc451579159c94eb48f2013

                       Task Name: exec
                          Caller: choria=romain.mcollective
                       Completed: 1
                         Running: 0

                      Successful: 1
                          Failed: 0

                Average Run Time: 0.16s


romain@optiplex-3050 ~ %
```

Fixes #98

:man_dancing:

### CODE REVIEW

File: lib/mcollective/util/tasks_support.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def platform_specific_command(path)
         # https://github.com/puppetlabs/pxp-agent/blob/3e7cada3cedf7f78703781d44e70010d0c5ad209/lib/src/modules/task.cc#L98-L107
         case extension
         when ".rb"
-          ["ruby", path]
+          ["C:/Program Files/Puppet Labs/Puppet/puppet/bin/ruby.exe", path]
```
Comment (smortex) [inline lib/mcollective/util/tasks_support.rb:118]: This has some side effect: we force using Ruby from AIO package and not another version setup by the administrator…  But this better match the behavior of Bolt (see #163 on that topic)…

Comment (ripienaar) [reply lib/mcollective/util/tasks_support.rb:118]: yeah agree thats the right thing - and also what unix is doing so seems best.

Comment (smortex) [reply lib/mcollective/util/tasks_support.rb:118]: I think so.  #163 will fix this as a side effect.  I'll amend this PR to not include the above change and send another PR to fix #163.

Comment (smortex) [reply lib/mcollective/util/tasks_support.rb]: Done in #166

**Comment by smortex** [comment]
The extra commit was removed.  This is ready for merging AFAIAC

**Comment by ripienaar** [comment]
thanks

**Pull request merged**